File: design-patterns/python/factory/simple_notification_factory.py
from .notification import Notification
from .email_notification import EmailNotification
from .sms_notification import SMSNotification
from .push_notification import PushNotification


class SimpleNotificationFactory:
    """Simple Factory Pattern"""
    
    @staticmethod
    def create_notification(notification_type: str) -> Notification:
        notification_type = notification_type.upper()
        
        if notification_type == "EMAIL":
            return EmailNotification()
        elif notification_type == "SMS":
            return SMSNotification()
        elif notification_type == "PUSH":
            return PushNotification()
        else:
            raise ValueError(f"Unknown notification type: {notification_type}")

File: design-patterns/java/factory/SimpleNotificationFactory.java
public class SimpleNotificationFactory {
    public static Notification createNotification(String type) {
        return switch (type) {
            case "EMAIL" -> new EmailNotification();
            case "SMS" -> new SMSNotification();
            case "PUSH" -> new PushNotification();
            default -> throw new IllegalArgumentException("Unknown type");
        };
    }    
}